test/net/sourceforge/kolmafia/preferences/PreferencesTest.java

```diff
@@ -2,59 +2,75 @@
 
 import static internal.helpers.Player.withProperty;
 import static internal.helpers.Player.withSavePreferencesToFile;
+import static internal.helpers.Utilities.verboseDelete;
 import static internal.matchers.Preference.isSetTo;
 import static org.hamcrest.CoreMatchers.containsString;
 import static org.hamcrest.CoreMatchers.not;
 import static org.hamcrest.MatcherAssert.assertThat;
-import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
 
 import internal.helpers.Cleanups;
 import java.io.File;
 import java.io.IOException;
+import java.io.InputStream;
 import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.nio.file.Paths;
 import java.util.TreeMap;
 import net.java.dev.spellcast.utilities.DataUtilities;
 import net.sourceforge.kolmafia.KoLCharacter;
+import net.sourceforge.kolmafia.KoLConstants;
+import net.sourceforge.kolmafia.KoLmafia;
 import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Nested;
 import org.junit.jupiter.api.Test;
 
 class PreferencesTest {
   private final String USER_NAME = "PreferencesTestFakeUser";
-  private final String EMPTY_USER = "Empty";
 
-  // These need to be before and after each because leakage has been observed between tests
-  // in this class.
   @BeforeEach
-  public void initializeCharPrefs() {
+  public void initializeCharPreferences() {
     KoLCharacter.reset(USER_NAME);
-    KoLCharacter.reset(true);
   }
 
   @AfterEach
-  public void resetCharAndPrefs() {
+  public void resetCharAndPreferences() {
+    deleteSerFiles(USER_NAME);
+    KoLmafia.releaseFileLock();
     KoLCharacter.reset("");
-    KoLCharacter.reset(true);
-    KoLCharacter.setUserId(0);
-    File userFile = new File("settings/" + USER_NAME.toLowerCase() + "_prefs.txt");
-    if (userFile.exists()) {
-      userFile.delete();
-    }
+  }
+
+  private static void deleteSerFiles(String username) {
+    String part = username.toLowerCase();
+    Path dest = Paths.get(KoLConstants.ROOT_LOCATION + "/data/" + part + "_queue.ser");
+    dest.toFile().delete();
+    dest = Paths.get(KoLConstants.ROOT_LOCATION + "/data/" + part + "_turns.ser");
+    dest.toFile().delete();
   }
 
   @Test
   void TestBackupFileWrite() {
+    // to test global prefs concurrency
+    String EMPTY_USER = "Empty";
     KoLCharacter.reset(EMPTY_USER);
     KoLCharacter.reset(true);
     KoLCharacter.setUserId(0);
     File userFile = new File("settings/" + EMPTY_USER.toLowerCase() + "_prefs.txt");
     File backupUserFile = new File("settings/" + EMPTY_USER.toLowerCase() + "_prefs.bak");
     if (userFile.exists()) {
-      userFile.delete();
+      if (!userFile.delete()) {
```

Comment:
why not `verboseDelete` as others in this file?

Reply:
verboseDelete was written as a debugging aid but as long as it exists I might as well use it.

Reply:
Done